translator/opencensus/oc_to_internal.go

```diff
@@ -0,0 +1,382 @@
+// Copyright 2019 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opencensus
+
+import (
+	"strings"
+
+	occommon "github.com/census-instrumentation/opencensus-proto/gen-go/agent/common/v1"
+	ocresource "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	octrace "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+	"github.com/golang/protobuf/ptypes"
+
+	"github.com/open-telemetry/opentelemetry-collector/consumer/consumerdata"
+	"github.com/open-telemetry/opentelemetry-collector/internal"
+	"github.com/open-telemetry/opentelemetry-collector/internal/data"
+	"github.com/open-telemetry/opentelemetry-collector/translator/conventions"
+	tracetranslator "github.com/open-telemetry/opentelemetry-collector/translator/trace"
+)
+
+func ocToInternal(td consumerdata.TraceData) data.TraceData {
```

Comment:
What about we do "OC -> OTLP -> Internal". We will need an "OTLP -> Internal" anyway for the otlp receiver so we reuse that code?

Reply:
I think we don't need to OC -> OTLP -> Internal, there is no need for such translation path. 

Here is what we need:
1. Right now we need OTLP->OC to implement OTLP receiver quickly without waiting for new internal data structures.
1. We need OC->OTLP to implement OTLP exporter quickly.

Once new data structures are ready we will need:
1. OpenCensus Receiver: OC->Internal
1. OpenCensus Exporter: Internal->OC
1. OTLP Receiver: OTLP->Internal
1. OTLP Exporter: Internal->OTLP

For compatibility shims we also need OC->Internal and Internal->OC.

Reply:
As you can see you propose "OC -> OTLP" in A2 and "OTLP -> Internal" in B3 if we implement these 2 we don't need B1 (OC -> Internal) because we can implement it as "OC -> OTLP -> Internal"

Reply:
Similar idea will apply for B2 we can have (B4 + A1). A - the first list; B - the second list

Reply:
I am not against this, just trying to reduce the code we add now and remove after the transition.

Reply:
The code in this PR will not need to be removed. It is needed for OpenCensus receiver.